packages/auth/src/grant/routes.ts

```diff
@@ -146,23 +146,29 @@ async function createGrant(
   deps: ServiceDependencies,
   ctx: CreateContext
 ): Promise<void> {
-  const isOnlyIncomingPaymentAccessRequest =
-    ctx.request.body.access_token.access
-      .map((acc) => {
-        return isIncomingPaymentAccessRequest(acc as IncomingPaymentRequest)
-      })
-      .every((el) => el === true)
-
-  if (
-    isOnlyIncomingPaymentAccessRequest &&
-    !deps.config.incomingPaymentInteraction
-  ) {
+  if (canSkipInteraction(deps, ctx)) {
     await createGrantInitiation(deps, ctx)
   } else {
     await createPendingGrant(deps, ctx)
   }
 }
 
+function canSkipInteraction(
+  deps: ServiceDependencies,
+  ctx: CreateContext
+): boolean {
+  const skipIncomingPaymentInteraction =
+    ctx.request.body.access_token.access.every(
+      isIncomingPaymentAccessRequest
+    ) && !deps.config.incomingPaymentInteraction
+
+  const skipQuoteInteraction =
+    ctx.request.body.access_token.access.every(isQuoteAccessRequest) &&
+    !deps.config.quoteInteraction
+
+  return skipIncomingPaymentInteraction || skipQuoteInteraction
```

Comment:
As is, this would return false for a grant request with only an incoming payment and quote access

Reply:
I originally decided against this because I wasn't sure of situations where you'd want to create an incoming payment and quote on the same payment pointer at a time, but in hindsight it shouldn't matter since both should be non-interactive... I'll update it